src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java

```diff
@@ -344,17 +344,16 @@ else if (abbrStarted) {
                 abbrStarted = false;
 
                 final int endIndex = index - 1;
-                // -1 as a first capital is usually beginning of next word
                 result = getAbbreviationIfIllegal(str, beginIndex, endIndex);
                 if (result != null) {
                     break;
                 }
                 beginIndex = -1;
             }
         }
-        // if abbreviation at the end of name and it is not single character (example: scaleX)
-        if (abbrStarted && beginIndex != str.length() - 1) {
-            final int endIndex = str.length();
+        // if abbreviation at the end of name (example: scaleX)
+        if (abbrStarted) {
+            final int endIndex = str.length() - 1;
```

Comment:
Since end isn't inclusive, shouldn't this be `endIndex = str.length()`?
Won't changing this get rid of `if (endIndex == str.length() - 1) {` in `getAbbreviation`?

Reply:
I agree it is confusion. It looks like a related update to the docs was just added to make this clearer: https://github.com/checkstyle/checkstyle/issues/5832
- `allowedAbbreviationLength`
- `allowedAbbreviations`

The two calls to `getAbbreviationIfIllegal(str, beginIndex, endIndex)` ensure that `str.substring(beginIndex, endIndex + 1)` is the complete sequence of consecutive upper-case characters.

That method compares `endIndex - beginIndex` to `allowedAbbreviationLength`, so it always drops one character off the end when doing this comparison, which I agree is confusing, but my understanding is this matches the existing behavior where `allowedAbbreviationLength` of `n` really means `n+1` characters are allowed.

Reply:
The two calls to `getAbbreviationIfIllegal(str, beginIndex, endIndex)` ensure that in both cases  `str.substring(beginIndex, endIndex + 1)` is a complete sequence of consecutive upper-case characters.

That method compares `endIndex - beginIndex` to `allowedAbbreviationLength`, so it always drops one character off the end when doing this comparison, which I agree is confusing, but my understanding is this matches the existing behavior where `allowedAbbreviationLength` of `n` really means `n+1` characters are allowed.

It looks like a related update to the docs was just added to make things clearer at the top: https://github.com/checkstyle/checkstyle/issues/5832

I think this is correct at least in the sense that it preserves the existing behavior for abbreviations not at the end of a string, and fixes it for abbreviations at the end of the string (where an additional character was dropped off the end).